# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gpe-appmgr/compile.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- gpe-appmgr-2.8/row_view.c	2005-01-05 22:14:47.000000000 +0100
+++ gpe-appmgr-2.8-fixed/row_view.c	2007-08-11 18:52:50.000000000 +0200
@@ -43,7 +43,6 @@
 /* GPE */
 #include <gpe/init.h>
 #include <gpe/pixmaps.h>
-#include <gpe/render.h>
 #include <gpe/spacing.h>
 #include <gpe/gpeiconlistview.h>
 #include <gpe/launch.h>
@@ -159,34 +159,33 @@
   gv = gdk_window_get_visual (programs_fixed->window);
   gcm = gdk_drawable_get_colormap (programs_fixed->window);
 
-  cr = cairo_create ();
   surface = cairo_xlib_surface_create (GDK_WINDOW_XDISPLAY (pixmap), GDK_WINDOW_XWINDOW (pixmap), 
 				       gdk_x11_visual_get_xvisual (gv), 0, 
 				       gdk_x11_colormap_get_xcolormap (gcm));
 
-  cairo_set_target_surface (cr, surface);
+  cr = cairo_create(surface);
 
-  cairo_set_rgb_color (cr, 1.0, 1.0, 1.0);
+  cairo_set_source_rgb (cr, 1.0, 1.0, 1.0);
   cairo_rectangle (cr, 0, 0, LABEL_WIDTH, height);
   cairo_fill (cr);
 
-  cairo_set_rgb_color (cr, 0xa5 / 255.0, 0xae / 255.0, 0xbf / 255.0);
+  cairo_set_source_rgb (cr, 0xa5 / 255.0, 0xae / 255.0, 0xbf / 255.0);
   cairo_move_to (cr, LABEL_WIDTH, 0);
   cairo_line_to (cr, 0, 0);
   cairo_line_to (cr, 0, height);
   cairo_line_to (cr, LABEL_WIDTH, height);
   cairo_stroke (cr);
 
-  cairo_select_font (cr, "Sans", CAIRO_FONT_SLANT_NORMAL,
+  cairo_select_font_face (cr, "Sans", CAIRO_FONT_SLANT_NORMAL,
 		     CAIRO_FONT_WEIGHT_NORMAL);
-  cairo_scale_font (cr, 14.0);
+  cairo_set_font_size (cr, 14.0);
 
   cairo_text_extents (cr, text, &extents);
 
   cairo_move_to (cr, 18, (height + extents.width) / 2);
 
   cairo_rotate (cr, -M_PI/2);
-  cairo_set_rgb_color (cr, 0, 0, 1.0);
+  cairo_set_source_rgb (cr, 0, 0, 1.0);
   cairo_show_text (cr, text);
 
   cairo_destroy (cr);
--- gpe-appmgr-2.8/main.c	2005-03-04 08:00:40.000000000 +0100
+++ gpe-appmgr-2.8-fixed/main.c	2007-08-11 18:29:46.000000000 +0200
@@ -50,7 +50,6 @@
 /* GPE */
 #include <gpe/init.h>
 #include <gpe/pixmaps.h>
-#include <gpe/render.h>
 #include <gpe/spacing.h>
 #include <gpe/gpeiconlistview.h>
 #include <gpe/gpeiconlistitem.h>
--- gpe-appmgr-2.8/single_view.c	2004-11-19 01:22:09.000000000 +0100
+++ gpe-appmgr-2.8-fixed/single_view.c	2007-08-11 18:31:25.000000000 +0200
@@ -39,7 +39,6 @@
 /* GPE */
 #include <gpe/init.h>
 #include <gpe/pixmaps.h>
-#include <gpe/render.h>
 #include <gpe/spacing.h>
 #include <gpe/gpeiconlistview.h>
 #include <gpe/tray.h>
--- gpe-appmgr-2.8/tab_view.c	2005-03-04 08:14:43.000000000 +0100
+++ gpe-appmgr-2.8-fixed/tab_view.c	2007-08-11 18:30:11.000000000 +0200
@@ -46,7 +46,6 @@
 /* GPE */
 #include <gpe/init.h>
 #include <gpe/pixmaps.h>
-#include <gpe/render.h>
 #include <gpe/spacing.h>
 #include <gpe/gpeiconlistitem.h>
 #include <gpe/gpeiconlistview.h>
